src/definitions/subscriptionField.ts

```diff
@@ -14,7 +14,7 @@ export function subscriptionField<FieldName extends string>(
     type: 'Subscription',
     definition(t) {
       const finalConfig = typeof config === 'function' ? config() : config
-      t.field(fieldName, finalConfig)
+      t.field(fieldName, finalConfig as any)
```

Comment:
Gets us around this issue for now:

![image](https://user-images.githubusercontent.com/284476/96508121-1f6a8b00-1228-11eb-8ce3-03f8e4de9f12.png)

Reply:
Maybe we should try to fix now.

Reply:
Found a fix, solution was to make `extendType` builder API polymorphic over the value of `TypeName`. Specifically branching over if it is `Subscription` or not.